kernel/Cargo.toml

```diff
@@ -89,7 +92,29 @@ default-engine = [
   "object_store",
   "parquet/async",
   "parquet/object_store",
-  "reqwest",
+  "reqwest/default",
+  "tokio",
+  "uuid/v4",
+  "uuid/fast-rng",
+]
+
+default-engine-rustls = [
```

Comment:
feature flags should be additive.

This should just be:
```toml
default-engine-rustls = [
  "default-engine",
  "reqwest/rustls-tls",
  "reqwest/charset",
  "reqwest/http2",
  "reqwest/macos-system-configuration",
]
```

Reply:
Ohh, I see, you _don't_ want `reqwest/default` in the case you specify `native-tls`.

I'd vote for your suggestion above then with `_default_engine_base`. Let's not maintain two independent but actually the same lists of deps

Reply:
I should have probably documented this better, but we want it to have:
- (old behavior) default-engine: reqwest with default features
- default-engine-rustls: reqwest _without_ default features, but with the list above

Reply:
yep, sounds good!

Reply:
Don't include OS specific deps, there are specific ways to only include it on the given OS.

Reply:
yea @sppalkia do we need `macos-system-configuration` flag?

Reply:
No clue -- I just copied and pasted the list of default features from `reqwest` ([link](https://github.com/seanmonstar/reqwest/blob/master/Cargo.toml#L30)). I'm okay removing it.

Reply:
hm I suppose let's remove and see if we can get along without it (though I am a little bit inclined to include it since it is part of default-features?) anyone have other thoughts?